backend/internal/daemon/docker_socket_proxy.go

```
@@ -299,10 +301,19 @@ func (p *dockerSocketProxy) serve(listener net.Listener, doneCh chan struct{}) {
 	}
 }
 
-// handle wakes the engine when needed, then bidirectionally proxies to the engine socket.
+// handle reads the CLI request first, then dials vsock. Dialing on accept
+// (before any HTTP) spends a guest connection on idle Compose/buildx sockets
+// and the virtio-vsock backend starts returning EOF on later _ping calls.
 func (p *dockerSocketProxy) handle(client net.Conn) {
 	defer client.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `client.Close` is not checked

(errcheck)